aptos-indexer-processors-sdk/sdk/src/common_steps/version_tracker_step.rs

```diff
@@ -0,0 +1,141 @@
+use crate::{
+    traits::{
+        pollable_async_step::PollableAsyncRunType, NamedStep, PollableAsyncStep, Processable,
+    },
+    types::transaction_context::TransactionContext,
+    utils::errors::ProcessorError,
+};
+use anyhow::Result;
+use async_trait::async_trait;
+use std::marker::PhantomData;
+
+#[async_trait]
+pub trait ProcessorStatusSaver {
+    // T represents the transaction type that the processor is tracking.
+    async fn save_processor_status(
+        &self,
+        tracker_name: &str,
+        last_success_batch: &TransactionContext<()>,
+    ) -> Result<(), ProcessorError>;
+}
+
+pub struct VersionTrackerStep<T, S>
```

Comment:
nit: add a comment about what this step does does, and we should include the assumption it makes about the data being ordered by version and if it's not ordered, use the `order` step